Gems/PhysX/Code/Source/EditorMeshColliderComponent.cpp

```diff
@@ -0,0 +1,1101 @@
+                        "PhysX Mesh",
+                        "Specifies the PhysX mesh collider asset for this PhysX collider component.")
+                    ->Attribute(AZ_CRC_CE("EditButton"), "")
+                    ->Attribute(AZ_CRC_CE("EditDescription"), "Open in Scene Settings")
+                    ->Attribute(AZ_CRC_CE("DisableEditButtonWhenNoAssetSelected"), true)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &EditorProxyAssetShapeConfig::m_configuration,
+                        "Configuration",
+                        "PhysX mesh asset collider configuration.")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &EditorProxyAssetShapeConfig::m_subdivisionLevel,
+                        "Subdivision level",
+                        "The level of subdivision if a primitive shape is replaced with a convex mesh due to scaling.")
+                    ->Attribute(AZ::Edit::Attributes::Min, Utils::MinCapsuleSubdivisionLevel)
+                    ->Attribute(AZ::Edit::Attributes::Max, Utils::MaxCapsuleSubdivisionLevel)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &EditorProxyAssetShapeConfig::ShowingSubdivisionLevel)
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorProxyAssetShapeConfig::OnConfigurationChanged);
+            }
+        }
+    }
+
+    EditorProxyAssetShapeConfig::EditorProxyAssetShapeConfig(
+        const Physics::PhysicsAssetShapeConfiguration& assetShapeConfiguration)
+        : m_configuration(assetShapeConfiguration)
+    {
+    }
+
+    bool EditorProxyAssetShapeConfig::ShowingSubdivisionLevel() const
+    {
+        return m_hasNonUniformScale;
+    }
+
+    AZ::u32 EditorProxyAssetShapeConfig::OnConfigurationChanged()
+    {
+        return AZ::Edit::PropertyRefreshLevels::ValuesOnly;
+    }
+
+    void EditorMeshColliderComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("PhysicsWorldBodyService"));
+        provided.push_back(AZ_CRC_CE("PhysicsColliderService"));
+        provided.push_back(AZ_CRC_CE("PhysicsTriggerService"));
+    }
+
+    void EditorMeshColliderComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("TransformService"));
+        required.push_back(AZ_CRC_CE("PhysicsRigidBodyService"));
+    }
+
+    void EditorMeshColliderComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+        dependent.push_back(AZ_CRC_CE("NonUniformScaleService"));
+    }
+
+    void EditorMeshColliderComponent::Reflect(AZ::ReflectContext* context)
+    {
+        EditorProxyAssetShapeConfig::Reflect(context);
```

Comment:
This file is basically a copy of `EditorColliderComponent` keeping everything related to PhysX Asset and removing anything to do with primitive shapes. This class now uses `EditorProxyAssetShapeConfig` which only configures a physx asset.

Reply:
I think as myself and @greerdv mentioned the Component Mode logic should now be removed from here too

Reply:
Editor Mesh Collider uses component mode to handle the offset, rotation and asset scale of the component.